sed/core/config.py

```diff
@@ -1,60 +1,39 @@
-"""This module contains a settings/config library for loading yaml/json files into
-dicts
-
+"""This module contains a config library for loading yaml/json files into dicts
 """
 import json
-import os
-from importlib.util import find_spec
 from pathlib import Path
 from typing import Union
 
 import yaml
 
-package_dir = os.path.dirname(find_spec("sed").origin)
+DEFAULT_CONFIG_PATH = Path(__file__).resolve().parent / "default.yaml"
 
 
-def parse_config(
-    config: Union[dict, str] = None,
-    default_config: Union[
-        dict,
-        str,
-    ] = f"{package_dir}/config/default.yaml",
-) -> dict:
-    """Load the config dictionary from a file, or pass the provided config dictionary.
+def parse_config(config: Union[dict, str] = None) -> dict:
+    """Load the default config dictionary, and update it if a user file or dict
+    is provided.
```

Comment:
This version is not reusable anymore. What I discussed with Steinn at DPG is that we want most likely for every scan to have a config file that is read automatically, and merged with a per-site config, that in-turn is merged with a per-package config. Thus the requirement of reuasability of this function....

Reply:
I reverted to a reusable version, and renamed it. It is indeed reused heavily in the new code.